### PR TITLE
Mod Manager KH 1.5+2.5 collection support and restore update

### PR DESCRIPTION
**Pull request by shananas**
Restore added to the mod loader for PC. Restore when used with patching also deletes the mod folder for the game selected so no extra mod unchecking is necessary with mods bundling scripts. Mod Loader Restore button does not restore packages just deletes the mod folder for the selected Launch Game.

Updates the setup wizard to support all 4 games in the collection with checkboxes and allowed the user to select next on game extraction page as long as one game is extracted not just kh2. To utilize this the Panacea changes Main Memory made will also need to be merged for the new mod folder structure. as now inside the mod folder a subfolder is made for the selected game in Launch Game. Patching only works with KH2 was unable to get it working with other games.

Visual bug when extracting games the file count from the header doesnt correctly add up to what the total files actualy are so the progress bar finishes before extraction is done at least when extracting all games. Did not test every combination of games.

### CODE REVIEW

File: OpenKh.Tools.ModsManager/Services/ModsService.cs [hunk 1]
```diff
@@ -167,9 +167,30 @@ public static IEnumerable<string> EnabledMods
...
             {
                 var _yamlGen = new Metadata();
-
-                _yamlGen.Title = modName + " (KH2PCPATCH)";
-                _yamlGen.Description = "This is an atuomatically generated metadata for this KH2PCPATCH Modification.";
+                if (fileName.Contains(".kh2pcpatch"))
+                {
+                    _yamlGen.Title = modName + " (KH2PCPATCH)";
+                    _yamlGen.Game = "kh2";
+                    _yamlGen.Description = "This is an atuomatically generated metadata for this KH2PCPATCH Modification.";
```
Comment (Delta-473) [inline OpenKh.Tools.ModsManager/Services/ModsService.cs:174]: atuomatically  typo ;)

File: OpenKh.Tools.ModsManager/Services/ModsService.cs [hunk 2]
```diff
@@ -167,9 +167,30 @@ public static IEnumerable<string> EnabledMods
...
+                }
+                else if (fileName.Contains(".kh1pcpatch"))
+                {
+                    _yamlGen.Title = modName + " (KH1PCPATCH)";
+                    _yamlGen.Game = "kh1";
+                    _yamlGen.Description = "This is an atuomatically generated metadata for this KH1PCPATCH Modification.";
```
Comment (Delta-473) [inline OpenKh.Tools.ModsManager/Services/ModsService.cs:180]: atuomatically  typo ;)

File: OpenKh.Tools.ModsManager/Services/ModsService.cs [hunk 3]
```diff
@@ -167,9 +167,30 @@ public static IEnumerable<string> EnabledMods
...
+                }
+                else if (fileName.Contains(".compcpatch"))
+                {
+                    _yamlGen.Title = modName + " (COMPCPATCH)";
+                    _yamlGen.Game = "recom";
+                    _yamlGen.Description = "This is an atuomatically generated metadata for this COMPCPATCH Modification.";
```
Comment (Delta-473) [inline OpenKh.Tools.ModsManager/Services/ModsService.cs:186]: atuomatically  typo ;)

File: OpenKh.Tools.ModsManager/Services/ModsService.cs [hunk 4]
```diff
@@ -167,9 +167,30 @@ public static IEnumerable<string> EnabledMods
...
+                }
+                else if (fileName.Contains(".bbspcpatch"))
+                {
+                    _yamlGen.Title = modName + " (BBSPCPATCH)";
+                    _yamlGen.Game = "bbs";
+                    _yamlGen.Description = "This is an atuomatically generated metadata for this BBSPCPATCH Modification.";
```
Comment (Delta-473) [inline OpenKh.Tools.ModsManager/Services/ModsService.cs:192]: atuomatically  typo ;)

**Review by Delta-473 — CHANGES_REQUESTED** [review]

File: OpenKh.Tools.ModsManager/Services/ModsService.cs (hunk 4 above)
Comment (shananas) [reply OpenKh.Tools.ModsManager/Services/ModsService.cs]: done

File: OpenKh.Tools.ModsManager/Services/ModsService.cs (hunk 3 above)
Comment (shananas) [reply OpenKh.Tools.ModsManager/Services/ModsService.cs]: done

File: OpenKh.Tools.ModsManager/Services/ModsService.cs (hunk 2 above)
Comment (shananas) [reply OpenKh.Tools.ModsManager/Services/ModsService.cs]: done

File: OpenKh.Tools.ModsManager/Services/ModsService.cs (hunk 1 above)
Comment (shananas) [reply OpenKh.Tools.ModsManager/Services/ModsService.cs]: done

**Pull request merged**